### DISCUSSION TITLE
Copilot replies by re-writing entire file

### DISCUSSION THREAD

**Discussion by gambithunt** [community/community#137735] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

There are many times where I highlight a piece of code and ask copilot something.

For example, if I highlight the code below and ask:

```
<p >
         Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
  </p>
```

Ask: How do I add a tailwind class so that there is a margin at the top of 20px?

Copilot will replay by rewriting the entire file of code with the change

```
<p class="mt-5">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        </p>
```

It would be so much quicker and easier to find the change if it re-wrote only what it changed.

**Comment by ForareVictor** [comment]
They ruined it with the last couple of updates, it is 100% worse than when I started using it, it fail to put any code in the project into context when giving answers and it is no longer possible to have a conversation with it about a solution it is like talking to a wall, it's just very robotic and shits out random code ignoring most of what we tried before or discussed, ( all this worked some months ago)
